src/registrar/models/models.py

```diff
@@ -83,3 +87,179 @@ def __str__(self):
+            "a tribal government recognized by the federal or " "state government",
+        ),
+        (COUNTY, "a county, parish, or borough"),
+        (CITY, "a city, town, township, village, etc."),
+        (SPECIAL_DISTRICT, "an independent organization within a single state"),
+    ]
+    organization_type = models.CharField(
+        max_length=255, choices=ORGANIZATION_CHOICES, help_text="Type of Organization"
+    )
+
+    EXECUTIVE = "Executive"
+    JUDICIAL = "Judicial"
+    LEGISLATIVE = "Legislative"
+    BRANCH_CHOICES = [(x, x) for x in (EXECUTIVE, JUDICIAL, LEGISLATIVE)]
+    federal_branch = models.CharField(
+        max_length=50,
+        choices=BRANCH_CHOICES,
+        null=True,
+        help_text="Branch of federal government",
+    )
+
+    is_election_office = models.BooleanField(
+        null=True, help_text="Is your ogranization an election office?"
+    )
+
+    organization_name = models.TextField(null=True, help_text="Organization name")
+    street_address = models.TextField(null=True, help_text="Street Address")
+    unit_type = models.CharField(max_length=15, null=True, help_text="Unit type")
+    unit_number = models.CharField(max_length=255, null=True, help_text="Unit number")
+    state_territory = models.CharField(
+        max_length=2, null=True, help_text="State/Territory"
+    )
+    zip_code = models.CharField(max_length=10, null=True, help_text="ZIP code")
+
+    authorizing_official = models.ForeignKey(
+        Contact,
+        null=True,
+        related_name="authorizing_official",
+        on_delete=models.PROTECT,
+    )
+
+    # "+" means no reverse relation to lookup applications from Website
+    current_websites = models.ManyToManyField(Website, related_name="current+")
+
+    requested_domain = models.ForeignKey(
+        Website,
+        null=True,
+        help_text="The requested domain",
+        related_name="requested+",
+        on_delete=models.PROTECT,
+    )
+    alternative_domains = models.ManyToManyField(Website, related_name="alternatives+")
+
+    submitter = models.ForeignKey(
+        Contact,
+        null=True,
+        related_name="submitted_applications",
+        on_delete=models.PROTECT,
+    )
+
+    purpose = models.TextField(null=True, help_text="Purpose of the domain")
+
+    other_contacts = models.ManyToManyField(
+        Contact, related_name="contact_applications"
+    )
+
+    security_email = models.CharField(
+        max_length=320, null=True, help_text="Security email for public use"
+    )
+
+    anything_else = models.TextField(
+        null=True, help_text="Anything else we should know?"
+    )
+
+    acknowledged_policy = models.BooleanField(
+        null=True, help_text="Acknowledged .gov acceptable use policy"
+    )
+
+    def can_submit(self):
+        """Return True if this instance can be marked as submitted."""
+        if not Website.string_could_be_domain(self.requested_domain):
+            return False
+        return True
```

Comment:
Error handling.

1. `requested_domain` can be null, but `Website.string_could_be_domain()` does not know how to deal with that.
2. `requested_domain` is a `Website` object, but `Website.string_could_be_domain()` does not know how to deal with that, either.

Also, nice demonstration of FSM. 🍪

Reply:
Refactored.